server/src/github-auth/github-auth.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller, Get, Req, UseGuards, Res } from '@nestjs/common';  
+import { AuthGuard } from '@nestjs/passport';
+import { GithubAuthService } from './github-auth.service';
+
+@Controller('auth/github')
+export class GithubAuthController {
+  constructor(
+    private readonly githubAuthService: GithubAuthService
+  ) {}
+
+  @Get()
+  @UseGuards(AuthGuard('github'))
+  async login() {
+    // initiates the GitHub OAuth2 login flow
+  }
+
+  @Get('callback')
+  @UseGuards(AuthGuard('github'))
+  async authCallback(@Req() req, @Res() res) {
+    const user = req.user;
+
+    try {
+      await this.githubAuthService.handleGithubAuthCallback(user);
+      return res.redirect('http://localhost:8081/Area');
```

Comment:
Can you use env var: if we have the time (I wish) to deploy this, we will maybe not gonna use this same ip / port.

Reply:
fixed but we also need to change on github OAuth app if we deploy